golden_hour/tweet.py

```diff
@@ -0,0 +1,68 @@
+from __future__ import absolute_import
+
+import argparse
+import logging
+import twitter
+import yaml
+
+import schema
+
+
+logger = logging.getLogger()
+
+def verify_credentials(credentials):
+    api = twitter.Api(**credentials)
+
+    return api.VerifyCredentials() is not None
+
+TWITTER_CONFIG_SCHEMA = schema.And(
+    {
+        'consumer_key': str,
+        'consumer_secret': str,
+        'access_token_key': str,
+        'access_token_secret': str,
+    },
+    verify_credentials
+)
+
+
+def post_update(credentials, text, media=None):
+    TWITTER_CONFIG_SCHEMA.validate(credentials)
+
+    logger.info('posting to twitter (status_text: {}, media: {})'.format(text, media))
+    api = twitter.Api(**credentials)
+
+    media_id = None
+    if media:
+        with open(media, 'rb') as mediafile:
+            media_id = api.UploadMediaChunked(mediafile)
+
+    api.PostUpdate(text, media=media_id)
+
+
+def load_credentials_from_file(filepath, nested=False):
+    ''' Load credentials from a YAML file.
+    If nested=True, expects the twitter secrets to be under a 'twitter' key in the file.
```

Comment:
Would it be easier to just check `'twitter' in conf`?

Reply:
why yes, yes it would.

Reply:
updated and tested manually with my nested config file